src/all/torrentio/src/eu/kanade/tachiyomi/animeextension/all/torrentio/Torrentio.kt

```diff
@@ -0,0 +1,715 @@
+    override fun popularAnimeRequest(page: Int): Request {
+        val variables = """
+            {
+                "page": $page,
+                "perPage": 30,
+                "sort": "POPULARITY_DESC"
+            }
+        """.trimIndent()
+
+        return makeGraphQLRequest(anilistQuery(), variables)
+    }
+
+    override fun popularAnimeParse(response: Response): AnimesPage {
+        val jsonData = response.body.string()
+        return parseSearchJson(jsonData) }
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int): Request {
+        val variables = """
+            {
+                "page": $page,
+                "perPage": 30,
+                "sort": "TIME_DESC"
+            }
+        """.trimIndent()
+
+        return makeGraphQLRequest(anilistLastestQuery(), variables)
+    }
+
+    override fun latestUpdatesParse(response: Response): AnimesPage {
+        val jsonData = response.body.string()
+        return parseSearchJson(jsonData, true)
+    }
+
+    // =============================== Search ===============================
+    override suspend fun getSearchAnime(page: Int, query: String, filters: AnimeFilterList): AnimesPage {
+        return if (query.startsWith(PREFIX_SEARCH)) { // URL intent handler
+            val id = query.removePrefix(PREFIX_SEARCH)
+            client.newCall(GET("$baseUrl/anime/$id"))
+                .awaitSuccess()
+                .use(::searchAnimeByIdParse)
+        } else {
+            super.getSearchAnime(page, query, filters)
+        }
+    }
+
+    private fun searchAnimeByIdParse(response: Response): AnimesPage {
+        val details = animeDetailsParse(response)
+        return AnimesPage(listOf(details), false)
+    }
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val variables = """
+            {
+                "page": $page,
+                "perPage": 30,
+                "sort": "POPULARITY_DESC",
+                "search": "$query"
+            }
+        """.trimIndent()
+
+        return makeGraphQLRequest(anilistQuery(), variables)
+    }
+
+    override fun searchAnimeParse(response: Response) = popularAnimeParse(response)
+    // =========================== Anime Details ============================
+
+    override fun animeDetailsParse(response: Response): SAnime = throw UnsupportedOperationException()
+
+    override suspend fun getAnimeDetails(anime: SAnime): SAnime = anime
+
+    // ============================== Episodes ==============================
+    override fun episodeListRequest(anime: SAnime): Request {
+        return GET("https://anime-kitsu.strem.fun/meta/series/anilist%3A${anime.url}.json")
+    }
+
+    override fun episodeListParse(response: Response): List<SEpisode> {
+        val responseString = response.body.string()
+        val json = Json { ignoreUnknownKeys = true }
```

Comment:
Create a class property for JSON with `private val json: Json by injectLazy()` instead

Reply:
Updated the code